subprojects/base-services/src/main/java/org/gradle/util/internal/TextUtil.java

```diff
@@ -39,6 +39,8 @@ public String apply(String input) {
         }
     };
     private static final Pattern NON_UNIX_LINE_SEPARATORS = Pattern.compile("\r\n|\r");
+    private static final Pattern A_SINGLE_QUOTE =  Pattern.compile("'");
```

Comment:
Note these do not cover the case the artifact name had a backslash - those would need to be escaped as well.

That makes me more confident we do not want to share an escaping implementation with Transformers. 
It may be the case we we would want to embed all that escaping in BuildScriptBuilder - as the needs are quite specific to code generation, whereas Transformers' escaping is used in producing space-safe command line parameters. In that case, maybe TextUtil is not the best place for the code-related escaping methods to live (same for TextUtilTest not being the best place to unit test them).